drain-service/drain_training_inferencing.py

```diff
@@ -33,46 +29,147 @@
+    if template:
+        log_data.anomaly_level = anomaly_level
+        log_data.template_cluster_id = cluster_id
+        log_data.template_matched = template
+        log_data.inference_model = "drain"
+        if template_log:
+            template_log_payload = Payload(log=template_log, template_matched=template, template_cluster_id=cluster_id, _id=str(cluster_id))
+            return True, template_log_payload
+        return True, None
+    return False, None
+
+
+async def consume_logs(incoming_logs_to_train_queue, update_model_logs_queue, batch_processed_queue, model_trained_queue):
     async def subscribe_handler(msg):
-        payload_data = msg.data.decode()
+        payload_data = msg.data
+        logs_payload_list = PayloadList()
+        logs_payload = logs_payload_list.parse(payload_data)
         await incoming_logs_to_train_queue.put(
-            pd.read_json(payload_data, dtype={"_id": object, "cluster_id": str})
+            logs_payload
         )
+    async def inferenced_subscribe_handler(msg):
+        payload_data = msg.data
+        logs_payload_list = PayloadList()
+        logs_payload = logs_payload_list.parse(payload_data)
+        await update_model_logs_queue.put(logs_payload)
 
-    async def anomalies_subscription_handler(msg):
-        anomalies_data = msg.data.decode()
-        logging.info("got anomaly payload")
-        await logs_to_update_es.put(pd.read_json(anomalies_data, {"_id": object, "cluster_id": str}))
+    await nw.subscribe(
+        nats_subject="batch_processed_workload",
+        payload_queue=batch_processed_queue
+    )
 
     await nw.subscribe(
-        nats_subject="preprocessed_logs",
-        payload_queue=None,
-        subscribe_handler=subscribe_handler,
+        nats_subject="model_inferenced_workload_logs",
+        nats_queue="workers",
+        payload_queue=update_model_logs_queue,
+        subscribe_handler=inferenced_subscribe_handler,
     )
 
     await nw.subscribe(
-        nats_subject="anomalies",
+        nats_subject="preprocessed_logs_workload",
         payload_queue=None,
-        subscribe_handler=anomalies_subscription_handler,
+        subscribe_handler=subscribe_handler,
     )
 
+    await nw.subscribe(nats_subject="workload_parameters", payload_queue=model_trained_queue)
+
+async def persist_model(batch_processed_queue, current_template_miner):
+    last_upload = time.time()
+    while True:
+        processed_payload = await batch_processed_queue.get()
+        current_time = time.time()
+        if current_time - last_upload >= 3600:
+            current_template_miner.save_state()
+            last_upload = time.time()
 
+async def reset_model(model_training_signal_queue, current_template_miner):
+    while True:
+        payload = await model_training_signal_queue.get()
+        current_template_miner.reset_model()
+        current_template_miner.save_state()
+
+async def inference_logs(incoming_logs_queue, workload_template_miner):
```

Comment:
can we rename this function? it's a bit misleading, as it's not really inferencing with drain model, but just try to match cached results.